titan/model.py

```diff
@@ -358,48 +362,112 @@ def agents_interact(self, rel: "ag.Relationship"):
             interaction = self.interactions[interaction_type]
             interaction.interact(self, rel)
 
-    def die_and_replace(self):
+    def exit(self):
+        if self.exits == {}:
+            return
+        for agent in self.pop.all_agents:
+            for strategy in self.params.enter_exit.values():
+                exit = self.params.classes.exit[strategy.exit_class]
+                if exit.ignore_incar and agent.incar.active:
+                    continue
+                if exit.exit_type == "age_out":
+                    # agent ages out of model
+                    if agent.age > exit.age:
+                        self.exits[strategy.exit_class].append(agent)
+                        # agent can only leave once, move to next agent
+                        break
+                elif exit.exit_type == "death":
```

Comment:
if we were willing to set a miniminum python version of 3.10, we could start using match-case statements instead of if/elif/breaks

https://stackoverflow.com/questions/66877130/what-is-the-syntactical-equivalent-to-switch-case-in-python/66877137#66877137